### PR TITLE
add winston logger

### PR DESCRIPTION
**Pull request by anjeyM**

### CODE REVIEW

File: task-3/src/shared/loggers/error-logger.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import * as winston from 'winston';
+import * as path from 'path';
+import dotenv from "dotenv";
+
+dotenv.config();
```
Comment (Kalita-Roman) [inline task-3/src/shared/loggers/error-logger.ts]: I have already seen `dotenv.config()` in [src/index.ts](https://github.com/anjeyM/NodeJsCourse/blob/homework5/task-3/src/index.ts#L14).
Is it needed use several times?

File: task-3/src/middleware/services-logger/services-logger.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+          const args = Array.prototype.slice.call(arguments, 1);
+          console.log(`Function called: ${this.name} with arguments: ${args}`);
+          const res = this.apply(self, args);
+          Function.prototype.call = newCall;
+          return res
+        }
+        Function.prototype.call = newCall;
+      })();
+    next();
+}
```
Comment (Kalita-Roman) [inline task-3/src/middleware/services-logger/services-logger.ts:19]: Is this service used somewhere?

File: task-3/src/index.ts [hunk 3]
```diff
@@ -19,14 +21,22 @@ app.use(cors());
 app.use(express.json());
 app.use("/api/users", usersRouter);
 app.use("/api/groups", groupsRouter);
-// app.use(errorHandler);
+app.use(errorHandler);
```
Comment (Kalita-Roman) [inline task-3/src/index.ts]: Shouldn't this error handler be defined as very last middleware?

**Review by Kalita-Roman — APPROVED** [review]

File: task-3/src/shared/loggers/error-logger.ts (hunk 1 above)
Comment (anjeyM) [reply task-3/src/shared/loggers/error-logger.ts]: Deleted

File: task-3/src/middleware/services-logger/services-logger.ts (hunk 2 above)
Comment (anjeyM) [reply task-3/src/middleware/services-logger/services-logger.ts:19]: It was a part of a homework, but after checking this functionality I was turned it off to reduce total amount of logs in the console.

File: task-3/src/index.ts (hunk 3 above)
Comment (anjeyM) [reply task-3/src/index.ts]: Moved to the ends

**Pull request merged**